teuthology/orchestra/run.py

```diff
@@ -240,24 +240,21 @@ def __eq__(self, value):
         return self.value == value
 
 
-def quote(args):
+def convert_args_list_to_str(args):
     """
-    Internal quote wrapper.
+    Convert list args to str args. While doing so, quote str instances to
+    disable special interpretation of characters and substitute Raw instances
+    with their actual values.
     """
-    def _quote(args):
-        """
-        Handle quoted string, testing for raw charaters.
-        """
-        for a in args:
-            if isinstance(a, Raw):
-                yield a.value
-            else:
-                yield pipes.quote(a)
-    if isinstance(args, list):
-        return ' '.join(_quote(args))
-    else:
+    if isinstance(args, str):
         return args
 
+    assert isinstance(args, list)
+
+    args = [pipes.quote(a) if isinstance(a, str) else a.value for a in args]
```

Comment:
Can we avoid mutability? In general is bad practice to reuse existing variables?
For the case, I guess it would be enough to return immediately without overriding args:
```
return ' '.join(pipes.quote(a) if isinstance(a, str) else a.value for a in args)
```

Reply:
i concur with Kyr. IMHO, the readability is not improved with this change.  i took more time parsing this oneliner. and `quote()` is a better name, i think.

Reply:
> i concur with Kyr. IMHO, the readability is not improved with this change. i took more time parsing this oneliner. and quote() is a better name, i think.

Okay, in that case it's best to close the PR.